types/preinstalled-snap.d.ts

```diff
@@ -0,0 +1,4 @@
+import type { PreinstalledSnap } from '@metamask/snaps-controllers';
+
+declare const value: PreinstalledSnap;
+export default value;
```

Comment:
This is nice. Ensures that the preinstalled-snap.json is correctly typed.

Although it does feel like we are shifting where we are doing assertions lol.

Reply:
I believe this will work in the metamask-extension only because `@metamask/snaps-controllers` package already exists there. However, in repos that do not have this package installed the these types will not work (and in some typescript projects this will cause in error).

The extension will use the types available in the version of `@metamask/snaps-controllers` it has installed; not the version that is in use in this package.

Not sure if this will be a problem, but I thought I'd mention it

One way around this may be to list `@metamask/snaps-controllers` as a `peerDependency` (and optionally set `peerDependenciesMeta` for the package to `optional: true`).

Of course, you could also set it as a regular `dependency`.

There is another solution which is a bit extreme for this scenario, use an additional compilation step to rollup the types. More information: https://medium.com/@martin_hotell/typescript-library-tips-rollup-your-types-995153cc81c7

I've tried them all in a complex project, and only found success when using `api-extractor`, though it is overkill and I don't recommend it for this simple use case.

Reply:
Urgh, this is all just to get 1 single type...
I don't think this overhead is worth it...

For now I'd rather ditch this definitions file and rely on how typescript resolves JSON types. Sure the type will be loose compared to the `PreinstalledSnap` type, but that is fine we can do the assertion on the extension side.